BE-1-init-application

Randayy opened the pull request.

Randayy committed:
Create .gitignore

Randayy committed:
Create App

Ghibliliker commented on App/.env.sample:
```diff
@@ -0,0 +1,3 @@
+SECRET_KEY=
```
u use all of them?

Randayy replied:
not now , but in future I can

Ghibliliker commented on App/main.py:
rename to "app"

Randayy replied:
renamed

Ghibliliker commented on App/main.py:
```diff
@@ -0,0 +1,11 @@
+app = FastAPI()
+
+@app.get("/")
+async def health_check():
```
move it from main

Randayy replied:
moved to routes/routes.py

Ghibliliker commented on requirements.txt:
```diff
@@ -0,0 +1,6 @@
+types-redis
+uvicorn
+httpx
+pytest
```
indicate versions

Randayy replied:
included

Ghibliliker commented on tests/test_main.py:
```diff
@@ -0,0 +1,13 @@
+from fastapi.testclient import TestClient
+from App.main import app
+
+client = TestClient(app)
```
add it like as fixture

Randayy replied:
added

Makkrich reviewed (changes requested):
All Pavlo's comments +
- Define whole project structure
- Split requirements into requirements.txt and requirements.dev.txt
- You dont use HOST and PORT env vars

Randayy commented:
@Fan4ik20 > * Define whole project structure
it will be okay ? or how it should look
app
├── __init__.py
├── core
├── db
├── main.py
├── routes
│   ├── __init__.py
│   └── routes.py
├── schemas
├── services
├── settings.py
└── utils

Randayy committed:
fixes

Randayy commented:
Fixed every Pavlo`s comment and that 

- Split requirements into requirements.txt and requirements.dev.txt

- You dont use HOST and PORT env vars

How should project structure looks like?

@Fan4ik20 @Ghibliliker @Artemooon @Fhockk

Fhockk commented:
![image](https://github.com/Randayy/FastAPI_App/assets/55735120/2b95176f-c440-4fe6-b5a7-c9e1b7033575)
The documentation contains a photo of what the project structure should look like, are there any specific questions?

Fhockk commented:
You have folder "App" it is an error. move the files from it to the root folder, and also create the missing folders and __init__ files inside

Randayy committed:
Added Structure

Randayy committed:
Deleted folder app

Fhockk commented on settings.py:
```diff
@@ -0,0 +1,7 @@
+load_dotenv()
+
+HOST = os.getenv("HOST")
+PORT = os.getenv("PORT")
```
please move it to class (check pydantic_settings)

Fhockk commented on settings.py:
Wrong directory and rename it, move it to app/core/config.py

Fhockk commented on __init__.py:
delete it, dont need in root directory

Fhockk commented on requirements.dev.txt:
if u split requirements to separate files, you need do in this way:
requirements.txt:
fastapi==x.x.x
pydantic_settings==x.x.x 
uvicorn==x.x.x
pydantic==x.x.x

requirements_dev.txt:
-r requirements.txt
pytest=x.x.x
httpx==x.x.x
Other stuff

Fhockk commented:
When i said delete "App" i didnt mean "app". You had both, but first was wrong

Randayy committed:
Changed Structure

Randayy committed:
fixed

Randayy commented:
added folder app,
added /core/config.py,
deleted __init__.py in root directory,
changed requirements.txt and requirements.dev.txt

Fhockk commented on app/tests/test_main.py:
```diff
@@ -0,0 +1,19 @@
+from fastapi.testclient import TestClient
+from utils.main import app
+
+@pytest.fixture
```
move fixture to conftest

Randayy replied:
moved

Fhockk commented on app/tests/test_main.py:
```diff
@@ -0,0 +1,19 @@
+            }
+    
+if __name__ == "__main__":
+    pytest.main()
```
why do u need that?

Fhockk commented on app/tests/__init__.py:
Move test directory from "app". They have one level

Randayy replied:
moved

Fhockk commented on app/utils/main.py:
move main outside utills func. its in app directory. And where do u start ur app with ur envs?

Randayy replied:
moved

Randayy committed:
fix

Fhockk commented on app/routers/routes.py:
its good to name in some logic way: health_check_route (for example)

Randayy replied:
renamed

Fhockk replied:
push commit then

Fhockk commented on tests/test_main.py:
```diff
@@ -0,0 +1,11 @@
+from .conftest import client
```
delete it, u dont need to import fixture if it initialized in conftest

Randayy replied:
deleted

Fhockk commented on app/main.py:
```diff
@@ -0,0 +1,8 @@
+app = FastAPI()
+
+app.include_router(router)
+
```
where is ur logic to start ur app by envs credentials?
and it is good practise to add DEBUG value to env and use it with "reload" value

Randayy replied:
added DEBUG to env and used it with reload

Fhockk reviewed (changes requested):
> Describe how your application starts in README.md

Randayy committed:
fixes

Fhockk commented:
> Describe how your application starts in README.md

still dont have it

Randayy commented:
made to run app use this:
```bash
python -m app.main
